pil-analyzer/src/condenser.rs

```diff
@@ -674,45 +678,190 @@ fn to_expr<T: Clone>(value: &Value<'_, T>) -> AlgebraicExpression<T> {
 
 /// Turns a value of function type (i.e. a closure) into a FunctionValueDefinition
 /// and sets the expected function kind.
-/// Does not allow captured variables.
-fn closure_to_function<T: Clone + Display>(
-    source: &SourceRef,
+/// Does allow some forms of captured variables.
+fn closure_to_function<T: FieldElement>(
+    _source: &SourceRef,
     value: &Value<'_, T>,
```

Comment:
I'm confused about this argument given the function name: is this expected to always be a closure?

Reply:
If so, how about `try_to_function_value_definition`?